contracts/NFTs/IntegrationId.sol

```diff
@@ -0,0 +1,48 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "./Base/NftBaseUpgradeable.sol";
+import "../interfaces/IDimoRegistry.sol";
+
+contract IntegrationId is Initializable, NftBaseUpgradeable {
+    IDimoRegistry private _dimoRegistry;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        string calldata name_,
+        string calldata symbol_,
+        string calldata baseUri_
+    ) external initializer {
+        _baseNftInit(name_, symbol_, baseUri_);
+    }
+
+    /// @notice Sets the DIMO Registry address
+    /// @dev Only an admin can set the DIMO Registry address
+    /// @param addr The address to be set
+    function setDimoRegistryAddress(address addr)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(addr != address(0), "Non zero address");
+        _dimoRegistry = IDimoRegistry(addr);
+    }
+
+    /// @notice Internal function to transfer a token
+    /// @dev Caller must have the transferer role
+    /// @dev The new owner must bet allowed to own an Integration node
+    /// @param from Old owner
+    /// @param to New owner
+    /// @param tokenId Token Id to be transferred
+    function _transfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal override {
+        _dimoRegistry.setIntegrationMinted(to);
```

Comment:
Can I ask what this is for? Is it a controller-type thing?

Reply:
Yes. Same idea of the Manufacturer to ensure each "Integrator" have just one node.

Reply:
I assume we can still mint a bunch into the same custody wallet, as for manufacturers.

The inheritance gets so confusing.

Reply:
Yep. I also updated this function setting the old owner as not minted since it does not possess a node anymore.